temporal/server_options.go

```diff
@@ -59,7 +60,8 @@ type (
 		persistenceServiceResolver resolver.ServiceResolver
 		elasticseachHttpClient     *http.Client
 		dynamicConfigClient        dynamicconfig.Client
-		customDataStoreFactory     client.AbstractDataStoreFactory
+		customDataStoreFactory     persistenceClient.AbstractDataStoreFactory
+		clientFactoryProvider      client.FactoryProvider
```

Comment:
customClientFactory?

Reply:
This is not factory itself, this is factory provider that provide a factory.